pkg/kapp/cmd/app/deploy.go

```diff
@@ -133,8 +134,13 @@ func (o *DeployOptions) Run() error {
 		return err
 	}
 
+	usedGVKs, err := app.UpdateUsedGVKs(NewUsedGVKsScope(newResources).GVKs(), true)
+	if err != nil {
+		return err
+	}
+
 	existingResources, existingPodRs, err := o.existingResources(
-		newResources, labeledResources, resourceFilter, supportObjs.Apps)
+		newResources, labeledResources, resourceFilter, supportObjs.Apps, usedGVKs)
```

Comment:
I think we can just pass app.UsedGVKs here because we are trying to get existing resources. We don't need to make function call to UpdateUsedGVKs in line 137.

Reply:
We need the function call on line 137 to track new GVKs that might have been introduced in the manifest.